actions/access_keys/search.go

```diff
@@ -51,3 +54,64 @@ func (a *Action) search(c *gin.Context) {
 
 	c.JSON(http.StatusOK, accessKeyContracts)
 }
+
+// search will fetch a list of access keys filtered by metadata
+// Search access key godoc
+// @Summary		Search access key
+// @Description	Search access key
+// @Tags		Access-key
+// @Produce		json
+// @Param		SearchAccessKeys body filter.SearchAccessKeys false "Supports targeted resource searches with filters and metadata, plus options for pagination and sorting to streamline data exploration and analysis"
+// @Success		200 {object} []response.AccessKey "List of access keys"
+// @Failure		400	"Bad request - Error while SearchAccessKeys from request body"
+// @Failure 	500	"Internal server error - Error while searching for access keys"
+// @Router		/api/v1//users/current/keys [get]
+// @Security	x-auth-xpub
+func (a *Action) search(c *gin.Context) {
+	reqXPubID := c.GetString(auth.ParamXPubHashKey)
+
+	searchParams, err := query.ParseSearchParams[filter.SearchAccessKeys](c)
```

Comment:
you should use `filter.AccessKeyFilter` instead of `filter.SearchAccessKeys`

Reply:
done